q_sort.py

```diff
@@ -0,0 +1,60 @@
+from merge_sort import timed_func
+
+def q_sort(sequence): 
+    if len(sequence) <= 1:
+        # Return a sequence of one sorted item, or stop if no input
+        return sequence
+    if len(sequence) == 2:
+        # Return a sequence of two sorted items
+        if sequence[1] > sequence[0]:
+            return sequence
+        else:
+            return sequence[::-1]
+    pivot = get_pivot(sequence)
+    left = []
+    right = []
+    not_pivot = False
+    for item in sequence:
+
+        if item > pivot[0] and item is not pivot[0]:
+            right.append(item)
+        elif item is not pivot[0] or not_pivot:
+            left.append(item)
+        elif item == pivot[0]:
+            # after not putting the first pivot in, put the rest in
+            not_pivot = True
```

Comment:
this feels risky to me.  I'm going to look for tests that ensure that with repeated values, the length of the list doesn't change.